CIP-XXXX/README.md

```diff
@@ -0,0 +1,205 @@
+---
+CIP: 92
+Title: First-class errors in Plutus
+Category: Plutus
+Status: Proposed
+Authors:
+    - Las Safin <me@las.rs>
+Implementors:
+    - Las Safin <me@las.rs>
+Discussions:
+    - https://github.com/cardano-foundation/cips/pulls/?
+Created: 2023-03-01
+License: CC-BY-4.0
+---
+
+## Abstract
+
+Plutus is a strictly evaluated language, where users heavily rely on **impure** errors
+to cause transaction failure to rule out invalid cases.
+
+We propose three changes:
+- Make `error` a first-class value you can pass around. Using it does not inherently
+  cause an error.
+- Change condition for script success from not erring to returning `True`.
+- Erase type abstractions and type applications entirely when going from
+  TPLC to UPLC rather than turning them into `delay` and `force`.
+
+UPLC is now **truly pure** wrt. transaction validation.
+This enables another use-case: We can **optimise** it as if it
+were entirely pure. We can turn `const () x` into `()` for any `x`!
+We can turn `expensive1 || expensive2` into `if expensive1 then True else expensive2`,
+and in general do any transformations that we'd do on a pure language without
+regard for side-effects.
+
+Issues fixed / PRs closed by this CIP:
+- https://github.com/input-output-hk/plutus/issues/4826
+- https://github.com/input-output-hk/plutus/issues/4114
+- https://github.com/input-output-hk/plutus/pull/4118
+- https://github.com/cardano-foundation/CIPs/pull/459
```

Comment:
(not a format correction, but an editorial question)

1. This is unclear whether these Plutus issues/PRs are closed to to this CIP method being followed (please confirm if so... I can't verify this from the issue/PR threads) or because their solutions share the premise of the CIP.  Can you change "closed by this CIP" to a statement or paragraph that explains the relationship unambiguously?
2. Once we decide where this is going we can also link @nielstron about the stalled https://github.com/cardano-foundation/CIPs/pull/459 to discuss whether candidate CIP-0091 should also be progressed (or deprecated).

Reply:
> Once we decide where this is going we can also link @nielstron about the stalled https://github.com/cardano-foundation/CIPs/pull/459 to discuss whether candidate CIP-0091 should also be progressed (or deprecated).

Michael proposed to merge it as Proposed and I don't see why we should do anything else to it. It seems reasonable, even though I highly doubt we're ever going to implement it, given that a very simple workaround exists.

Reply:
@effectfully I think since this is doable, but not likely to be done, it _would_ make sense to put it through as Proposed after some cleanup from @L-as... rather than assign the currently not-yet-used `Rejected` status that e.g. Michael proposed for stalled candidate https://github.com/cardano-foundation/CIPs/pull/336 (though a `Rejected` status could still be considered if everyone is in agreement about that).

Reply:
@rphair sorry, I was commenting about #459 specifically. If my suspicion regarding this CIP not being doable is confirmed (if!), then this CIP is a prime candidate for being rejected.